dolphinscheduler-worker/src/main/java/org/apache/dolphinscheduler/server/worker/processor/TaskKillProcessor.java

```diff
@@ -94,8 +94,14 @@ public void process(Channel channel, Command command) {
         }
 
         int processId = taskExecutionContext.getProcessId();
-        if (processId == 0) {
-            this.cancelApplication(taskInstanceId);
+
+        this.cancelApplication(taskInstanceId);
+
+        Pair<Boolean, List<String>> result = doKill(taskExecutionContext);
```

Comment:
I don't think we need to do this change, if the task is execute in worker's thread, the processId is 0.

Reply:
+1